File: 08-contar-caracteres/java/ContarCaracteres.java
class ContarCaracteres {
    public static void main(String[] args) {
        String cadena = System.console().readLine("Introduce una cadena de caracteres: ");
        char caracter = System.console().readLine("Introduce un caracter: ").charAt(0);
        int contador = 0;
        for (int i = 0; i < cadena.length(); i++) {
            if (cadena.charAt(i) == caracter) {
                contador++;
            }
        }
        System.out.println("El caracter " + caracter + " aparece " + contador + " veces en la cadena");
    }
}

File: 08-contar-caracteres/php/contar_caracteres.php
<?php

$texto = readline("Introduzca un texto: ");
$caracter = readline("Introduzca un caracter: ");

$contador = 0;
for ($i = 0; $i < strlen($texto); $i++) {
    if ($texto[$i] == $caracter) {
        $contador++;
    }
}

echo "El caracter $caracter aparece $contador veces en el texto";